pegasus/sites.v3/code.org/views/census_map.haml

```diff
@@ -81,9 +104,56 @@
 
     google.maps.event.addListener(layer, 'click', function(event) {
       enableDrag();
-      event.infoWindowHtml = event.row['organization_name'].value;
-      event.infoWindowHtml += ' (' + event.row['city'].value + ')';
-      event.infoWindowHtml += '<br>';
+      let censusMessage;
+      switch(event.row['teaches_cs'].value) {
+        case 'YES':
+          censusMessage = "We believe this school offers Computer Science class(es).";
+          break;
+        case 'NO':
+          censusMessage = "We believe this school offers limited or no Computer Science opportunities.";
+          break;
+        case 'MAYBE':
+          censusMessage = "We are unsure if this school offers Computer Science class(es).";
+          break;
+        default:
+          censusMessage = "We have no data about this school.";
+      }
+      const schoolDropdownOptionString = `{
+        value: '${event.row['school_id'].value}',
+        label: '${event.row['school_name'].value}',
+        school: {
+          nces_id: '${event.row['school_id'].value}',
+          name: '${event.row['school_name'].value}',
+          city: '${event.row['city'].value}',
+          state: '${event.row['state_code'].value}',
+          latitude: ${event.row['location'].value.split(',')[0]},
+          longitude: ${event.row['location'].value.split(',')[1]},
+        },
+      }`;
+      event.infoWindowHtml  = '<div class="census-info-window">';
+      event.infoWindowHtml += '  <h4>';
+      event.infoWindowHtml += '    <b>';
+      event.infoWindowHtml +=        event.row['school_name'].value;
+      event.infoWindowHtml += '    </b>';
+      event.infoWindowHtml += '    <br>';
+      event.infoWindowHtml += '    (' + event.row['city'].value + ', ' + event.row['state_code'].value +')';
+      event.infoWindowHtml += '  </h4>';
+      event.infoWindowHtml += '  <hr>';
+      event.infoWindowHtml += '  <div class="census-message">' + censusMessage + '</div>';
+      event.infoWindowHtml += '  <div class="button-container">';
+      event.infoWindowHtml += '    <a href="#form" onclick="showYourSchoolFunction(' + schoolDropdownOptionString + ');">';
+      event.infoWindowHtml += '      <div class="button">';
+      event.infoWindowHtml += '        <div class="button-text">Take the survey for this school</div>';
+      event.infoWindowHtml += '      </div>';
+      event.infoWindowHtml += '    </a>';
+      event.infoWindowHtml += '    <a href="TODO: add this link once we have it">';
```

Comment:
Are we okay with this going out as a TODO?

Reply:
This won't go out as a TODO. I just got the link from Miranda and will add it today.

Reply:
The link is fixed now.